core/src/main/java/org/apache/iceberg/util/LockManagers.java

```diff
@@ -54,6 +55,15 @@ public static LockManager from(Map<String, String> properties) {
     }
   }
 
+  public static LockManager from(DynMethods.UnboundMethod setVersionId, Map<String, String> properties) {
```

Comment:
I think we don't need to do this in the core library, we can just apply the logic in the GlueCatalog when initializing the catalog. This is too specific implementation detail for the GlueCatalog.

Reply:
Agreed!

Reply:
Thanks, I have made the changes.